### PR TITLE
Run clang-format on the step-49 example code.

### PR DESCRIPTION
**Pull request by drwells**
I also removed some C arrays in favor of `std::vector` while I was there.

### CODE REVIEW

File: examples/step-49/doc/results.dox [hunk 1]
```diff
@@ -74,170 +74,152 @@ third direction, and gluing them together. The following code does this, using
...
-       {4, 5, 8, 9},
-       {5, 6, 9, 10},
-       {7,11, 6,10},
-       {8, 9, 14,12},
-       {9, 10,12,13},
-       {11,15,10,13},
-       {14,12,15,13}
-  };
+  std::vector<Point<2>> vertices_1{Point<2>(-1.5, 0.),
+                                   Point<2>(-0.5, 0.),
```
Comment (bangerth) [inline examples/step-49/doc/results.dox]: I am surprised this actually works with just one set of braces. The braces here enclose constructor arguments, so I would have expected that you need a second set of braces to group all of these points to an initializer-list. Or that you write
```
  std::vector<...> vertices_1
  = { ...list of points ...};
```
We could also have made the things even simpler by replacing `Point<2>(-1.5,0)` to `{-1.5, 0}` and let the compiler do the conversion to `Point<2>` itself.

File: examples/step-49/doc/results.dox [hunk 2]
```diff
@@ -74,170 +74,152 @@ third direction, and gluing them together. The following code does this, using
...
+                       {{9, 10, 12, 13}},
+                       {{11, 15, 10, 13}},
+                       {{14, 12, 15, 13}}};
 
   // Copy vertices into a 2d triangulation
   Triangulation<2> triangulation_2d_1;
-  create_2d_grid (vertices_1,
-                  sizeof(vertices_1)/sizeof(vertices_1[0]),
-		  cell_vertices_1,
-                  sizeof(cell_vertices_1)/sizeof(cell_vertices_1[0]),
```
[the remark this answers is not in the document]
Comment (bangerth) [reply examples/step-49/doc/results.dox]: :+1:

File: examples/step-49/doc/results.dox (hunk 1 above)
Comment (drwells) [reply examples/step-49/doc/results.dox]: I try not to ask too many questions about how the curly bracket initialization works.

Comment (bangerth) [reply examples/step-49/doc/results.dox]: Yes, confusing indeed.

**Pull request merged**